SCAutolib/controller.py

```diff
@@ -221,136 +253,132 @@ def setup_user(self, user_dict: dict, force: bool = False):
                 new_user.delete_user()
-            user_dict["card_dir"].mkdir(exist_ok=True)
             new_user.add_user()
 
-            csr_path = new_user.card_dir.joinpath(f"csr-{new_user.username}.csr")
-            cnf = file.OpensslCnf(filepath=csr_path, conf_type="user",
-                                  replace=new_user.username)
-            cnf.create()
-            cnf.save()
-
-            new_user.cnf = cnf.path
-            self.sssd_conf.set(
-                section=f"certmap/shadowutils/{new_user.username}",
-                key="matchrule",
-                value=f"<SUBJECT>.*CN={new_user.username}.*")
-            self.sssd_conf.save()
-            self.sssd_conf.update_default_content()
-            run(["systemctl", "restart", "sssd"])
-            logger.debug(f"Match rule for user {new_user.username} is added "
-                         f"to /etc/sssd/sssd.conf")
         else:
             if self.ipa_ca is None:
                 msg = "Can't proceed in configuration of IPA user because no " \
                       "IPA Client is configured"
                 raise exceptions.SCAutolibException(msg)
             new_user = user.IPAUser(ipa_server=self.ipa_ca,
                                     username=user_dict["name"],
-                                    pin=user_dict["pin"],
-                                    password=user_dict["passwd"],
-                                    card_dir=card_dir,
-                                    cert=user_dict["cert"],
-                                    key=user_dict["key"],
-                                    local=False)
+                                    password=user_dict["passwd"])
             if force:
-                if new_user.cert and new_user.cert.exists():
-                    self.ipa_ca.revoke_cert(new_user.cert)
                 new_user.delete_user()
-            user_dict["card_dir"].mkdir(exist_ok=True)
-
             new_user.add_user()
+        self.users.append(new_user)
+        dump_to_json(new_user)
+        return new_user
 
-        if user_dict["card_type"] == "virtual":
-            hsm_conf = file.SoftHSM2Conf(
-                new_user.card_dir.joinpath("sofhtsm2.conf"),
-                new_user.card_dir)
+    def setup_card(self, card_dict: dict, force: bool = False):
+        """
+        Create card object. Card object should contain its root CA cert as it
+        represents general card (i.e. including physical read-only cards).
+
+        :param card_dict: Dictionary containing card attributes
+        :type card_dict: dict
+        :param force: If its true and card directory exists it will be removed
+        :type force: bool
+        """
+        card_dir: Path = Path("/root/cards", card_dict["name"])
+        card_dir.mkdir(parents=True, exist_ok=True)
+
+        if force and card_dir.exists():
+            rmtree(card_dir)
+
+        if card_dict["card_type"] == "physical":
```

Comment:
better to put this hardcoed string `"physiscal"` to the enum like
```python
class CardType(str, Enum):
   physical = "physiscal"
   virtual = "virtual"
```
and then use these values like `CardType.physical` because this approach would make your code more readable, explictely specified types, easy-to-change types